tests/test_modeling_rag.py

```diff
@@ -0,0 +1,895 @@
+            decoder_start_token_id=self.decoder_start_token_id,
+            max_combined_length=self.max_combined_length,
+            retrieval_vector_size=self.retrieval_vector_size,
+            retrieval_batch_size=self.retrieval_batch_size,
+        )
+
+        # BART params
+        self.hidden_size = 16
+        self.num_hidden_layers = 2
+        self.num_attention_heads = 4
+        self.intermediate_size = 4
+        self.hidden_dropout_prob = 0.1
+        self.attention_probs_dropout_prob = 0.1
+        self.max_position_embeddings = 20
+
+        self.bart_config = BartConfig(
+            vocab_size=self.vocab_size,
+            d_model=self.hidden_size,
+            encoder_layers=self.num_hidden_layers,
+            decoder_layers=self.num_hidden_layers,
+            encoder_attention_heads=self.num_attention_heads,
+            decoder_attention_heads=self.num_attention_heads,
+            encoder_ffn_dim=self.intermediate_size,
+            decoder_ffn_dim=self.intermediate_size,
+            dropout=self.hidden_dropout_prob,
+            attention_dropout=self.attention_probs_dropout_prob,
+            max_position_embeddings=self.max_position_embeddings,
+            eos_token_id=self.eos_token_id,
+            bos_token_id=self.bos_token_id,
+            pad_token_id=self.pad_token_id,
+            decoder_start_token_id=self.decoder_start_token_id,
+        )
+
+        # DPR params
+        self.dpr_vocab_size = 51
+        self.hidden_size = 20
+        self.num_hidden_layers = 3
+        self.num_attention_heads = 5
+        self.intermediate_size = 5
+        self.hidden_act = "gelu"
+        self.hidden_dropout_prob = 0.2
+        self.attention_probs_dropout_prob = 0.2
+        self.max_position_embeddings = 19
+        self.type_vocab_size = 17
+        self.initializer_range = 0.02
+        self.projection_dim = 0
+
+        self.dpr_config = DPRConfig(
+            projection_dim=self.projection_dim,
+            vocab_size=self.dpr_vocab_size,
+            hidden_size=self.hidden_size,
+            num_hidden_layers=self.num_hidden_layers,
+            num_attention_heads=self.num_attention_heads,
+            intermediate_size=self.intermediate_size,
+            hidden_act=self.hidden_act,
+            hidden_dropout_prob=self.hidden_dropout_prob,
+            attention_probs_dropout_prob=self.attention_probs_dropout_prob,
+            max_position_embeddings=self.max_position_embeddings,
+            type_vocab_size=self.type_vocab_size,
+            is_decoder=False,
+            initializer_range=self.initializer_range,
+        )
+
+    def prepare_inputs(self):
+        input_ids = ids_tensor([self.batch_size, self.seq_length], self.vocab_size).clamp(
+            3,
+        )
+        input_ids[:, -1] = self.eos_token_id
+        attention_mask = input_ids.ne(self.pad_token_id)
+
+        return input_ids, attention_mask
+
+
+@require_torch
+@require_retrieval
+class RagTestMixin:
```

Comment:
these look like common tests but don't inherit? Comment if intentional.

Reply:
Just like `EncoderDecoderModel`, the `RagModel` is too speciifc to inherit from the common tests => It requires a retriever to work which is not a given in the common tests.